File: Caelan.Frameworks.Common/Helpers/AssemblyHelper.fs
namespace Caelan.Frameworks.Common.Helpers

open System
open System.Reflection
open System.Web

module AssemblyHelper =
    let GetWebEntryAssembly() =
        if HttpContext.Current = null || HttpContext.Current.ApplicationInstance = null then
            null
        else
            let mutable t = HttpContext.Current.ApplicationInstance.GetType()
            while t <> null && t.Namespace = "ASP" do
                t <- t.BaseType
            
            match t with
            | null -> null
            | _ -> t.Assembly

File: Caelan.Frameworks.Common/Helpers/AssemblyHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Reflection;
using System.Text;
using System.Threading.Tasks;
using System.Web;

namespace Caelan.Frameworks.Common.Helpers
{
    public static class AssemblyHelper
    {
        public static Assembly GetWebEntrAssembly() => GetWebAssembly(HttpContext.Current?.ApplicationInstance?.GetType());

        private static Assembly GetWebAssembly(Type type) => type?.Namespace != "ASP" ? type?.Assembly : GetWebAssembly(type.BaseType);
    }
}
